python/lsst/ip/isr/isrFunctions.py

```diff
@@ -865,3 +865,39 @@ def setBadRegions(exposure, badStatistic="MEDIAN"):
     imageArray[:] = numpy.where(badPixels, value, imageArray)
 
     return badPixels.sum(), value
+
+
+def checkFilter(exposure, filterList, log=None):
+    """Check to see if an exposure is in a filter specified by a list.
+
+    The goal of this is to provide a unified filter checking interface
+    for all filter dependent stages.
+
+    Parameters
+    ----------
+    exposure : `lsst.afw.image.Exposure`
+        Exposure to examine.
+    filterList : `list` [`str`]
+        List of physical_filter names to check.
+    log : `lsst.log.Log`, optional
+        Logger to handle messages.
```

Comment:
This is not optional below in the first `if` branch. Given that we probably do want those warnings, I think it should be non-optional.

Reply:
Made it non-optional, as both current uses (fringe and straylight) have their own loggers to pass in.